### PR TITLE
add ability to set codepage explicitly for BIFF5

### PR DESCRIPTION
**Pull request by boltunoreh**
If BIFF5 excel 95 file doesn't have codepage record, the default codepage CP1252 is used and can't be change.
That causes to problems with decoding cyrillic text.

This is:

```
- [ ] a bugfix
- [x] a new feature
```

Checklist:

- [ ] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [ ] Documentation is updated as necessary

### Why this change is needed?
It's made to add ability to read Excel 95 files with cyrillic text properly.

### CODE REVIEW

File: src/PhpSpreadsheet/Reader/Xls.php [hunk 1]
```diff
@@ -439,6 +439,14 @@ public function canRead($pFilename)
         }
     }
 
+    /**
+     * @param string $codepage
+     */
+    public function setCodepage($codepage): void
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Reader/Xls.php:445]: You might want to consider typehinting, and some form of validation here

File: src/PhpSpreadsheet/Reader/Xls.php [hunk 2]
```diff
@@ -640,7 +648,7 @@ public function load($pFilename)
 
         // initialize
         $this->pos = 0;
-        $this->codepage = 'CP1252';
+        $this->codepage = $this->codepage == null ? 'CP1252' : $this->codepage;
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Reader/Xls.php:651]: Perhaps use `$this->codepage = ?: 'CP1252'` here.
As our minimum PHP version is 7.1, using more succinct syntax is available to us

**Comment by MarkBaker** [comment]
Also, take a look at the most recent merges to master, one of which addresses codepages

File: src/PhpSpreadsheet/Reader/Xls.php (hunk 1 above)
Comment (boltunoreh) [reply src/PhpSpreadsheet/Reader/Xls.php]: > You might want to consider typehinting, and some form of validation here

done

File: src/PhpSpreadsheet/Reader/Xls.php (hunk 2 above)
Comment (boltunoreh) [reply src/PhpSpreadsheet/Reader/Xls.php]: done

**Comment by boltunoreh** [comment]
> Also, take a look at the most recent merges to master, one of which addresses codepages

done

File: src/PhpSpreadsheet/Shared/CodePage.php [hunk 3]
```diff
@@ -65,13 +67,29 @@ class CodePage
...
 
+    /**
+     * @param string $codePage
+     *
+     * @return bool
+     */
+    public static function validate(string $codePage): bool
+    {
+        if (!in_array($codePage, self::$pageArray)) {
+            return false;
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Shared/CodePage.php:78]: Can be much simplified to a single line as `return (in_array($codePage, self::$pageArray, true));` with strict type checking for the codepage string as well

Comment (boltunoreh) [reply src/PhpSpreadsheet/Shared/CodePage.php]: fixed

File: src/PhpSpreadsheet/Shared/CodePage.php [hunk 4]
```diff
@@ -65,13 +67,25 @@ class CodePage
...
     ];
 
+    /**
+     * @param string $codePage
+     *
+     * @return bool
+     */
+    public static function validate(string $codePage): bool
+    {
+        return (in_array($codePage, self::$pageArray, true));
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Shared/CodePage.php:77]:
```suggestion
        return in_array($codePage, self::$pageArray, true);
```

File: src/PhpSpreadsheet/Reader/Xls.php [hunk 5]
```diff
@@ -439,6 +439,20 @@ public function canRead($pFilename)
         }
     }
 
+    /**
+     * @param string $codepage
+     *
+     * @throws PhpSpreadsheetException
+     */
+    public function setCodepage(string $codepage): void
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Reader/Xls.php:447]:
```suggestion
    public function setCodepage(string $codepage): void
```

File: src/PhpSpreadsheet/Shared/CodePage.php [hunk 6]
```diff
@@ -65,13 +67,25 @@ class CodePage
...
      * Convert Microsoft Code Page Identifier to Code Page Name which iconv
      * and mbstring understands.
      *
      * @param int $codePage Microsoft Code Page Indentifier
      *
      * @return string Code Page Name
+     *
+     * @throws PhpSpreadsheetException
      */
     public static function numberToName(int $codePage): string
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Shared/CodePage.php:90]:
```suggestion
    /**
     * Convert Microsoft Code Page Identifier to Code Page Name which iconv
     * and mbstring understands.
     *
     * @param int $codePage Microsoft Code Page Indentifier
     *
     * @return string Code Page Name
     */
    public static function numberToName(int $codePage): string
```

File: src/PhpSpreadsheet/Shared/CodePage.php [hunk 7]
```diff
@@ -65,13 +67,25 @@ class CodePage
         65001 => 'UTF-8', //    Unicode (UTF-8)
     ];
 
+    /**
+     * @param string $codePage
+     *
+     * @return bool
+     */
+    public static function validate(string $codePage): bool
+    {
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Shared/CodePage.php:76]:
```suggestion
    public static function validate(string $codePage): bool
    {
```

**Pull request merged**